### ISSUE TITLE
Can I execute inspec test on local and on containers ine the same script ?

### ISSUE THREAD

**Issue by Azaroths** [inspec/inspec#1762]
Hello,

I would like to know if it's possible to execute an inspec test on local and on containers ine the same script ?

Regards,

Azaroths

**Comment by chris-rock** [comment]
@Azaroths Could you please explain the use case?

**Comment by Azaroths** [comment]
I have to test commands in local and command on containers
I have a shell script who start script on containers but I would like this script executes test on local server too 

```
inspec exec container1.rb -t docker://"$container1"
inspec exec container2.rb -t docker://"$container2"
inspec exec container3.rb -t docker://"$container3"
inspec exec container4.rb -t docker://"$container4"
```

is it possible to execute test in the same script shell directly like this ->

```
# Execute tests on containers

inspec exec container1.rb -t docker://"$container1"
inspec exec container2.rb -t docker://"$container2"
inspec exec container3.rb -t docker://"$container3"
inspec exec container4.rb -t docker://"$container4"

# Execute tests directly on local

describe command('docker logs mycontainer') do
             its('stdout') { should match /GET/ }
end

describe command('docker ps name=mycontainer | grep healthy') do
            its('stdout') { should match /healthy/ }
end

```

**Comment by clintoncwolfe** [comment]
As implemented in InSpec 1.0 and InSpec 2, each `inspec exec` run is bound to exactly one "target".  This has consequences for many things, such as setting up a communications channel, identifying a platform, and skipping controls that do not support the target platform.

Here, you're looking for `inspec exec` to switch between different docker targets, then switch to a different target type (`local`) within one run.  

Making InSpec able to switch targets mid-run is conceivable, but would be a difficult (and likely brittle) change.  If I were you, I'd simply write a shell (or Perl, or whatever) script, which calls out to `inspec exec` 5 times with 5 different targets.  You can even get clever and parallelize them!

**Comment by clintoncwolfe** [comment]
As this has been idle for a while, I'm closing this. If you have any further questions on this topic, please re-open the ticket.
